src/Model/test.js

```diff
@@ -279,6 +263,96 @@ describe('watermelondb/Model', () => {
       }),
     ).rejects.toBeInstanceOf(Error)
   })
+
+  it('disallows operations on uncommited records', async () => {
+    const database = makeDatabase()
+    const model = MockModel._prepareCreate(database.collections.get('mock'), () => {})
+    expect(model._isCommitted).toBe(false)
+
+    await expectToRejectWithMessage(model.update(() => {}), /uncommitted/)
+    await expectToRejectWithMessage(model.markAsDeleted(), /uncommitted/)
+    await expectToRejectWithMessage(model.destroyPermanently(), /uncommitted/)
+    expect(() => model.observe()).toThrowError(/uncommitted/)
+  })
+  it('disallows changes on records with pending updates', async () => {
+    const database = makeDatabase()
+    database.adapter.batch = jest.fn()
+
+    const model = new MockModel(database.collections.get('mock'), {})
+    model.prepareUpdate()
+    expect(() => {
+      model.prepareUpdate()
+    }).toThrowError(/pending update/)
+    await expectToRejectWithMessage(model.update(() => {}), /pending update/)
+
+    // need to call batch or a dev check will get angry
+    database.batch(model)
+  })
+  it('disallows writes outside of an action', async () => {
+    const database = makeDatabase({ actionsEnabled: true })
+    database.adapter.batch = jest.fn()
+
+    const model = await database.action(() => database.collections.get('mock').create())
+
+    await expectToRejectWithMessage(
+      model.update(noop),
+      /can only be called from inside of an Action/,
+    )
+
+    await expectToRejectWithMessage(
+      model.markAsDeleted(),
+      /can only be called from inside of an Action/,
+    )
+
+    await expectToRejectWithMessage(
+      model.markAsDeleted(),
+      /can only be called from inside of an Action/,
+    )
+
+    // check that no throw inside action
+    await database.action(async () => {
+      await model.update(noop)
+      await model.markAsDeleted()
+      await model.markAsDeleted()
```

Comment:
I think one of them should be `destroyPermanently` :)

Reply:
nice catch!